drivers/gp8xxx: add initial support

basilfx opened the pull request:
### Contribution description
This adds support for the Guestgood GP8xxx I2C voltage and current DACs. The following variations are supported.

* GP8503: 2-channel 12-bit VDAC with 0-2.5/VCC V output range
* GP8512: 1-channel 15-bit VDAC with 0-2.5/VCC V output range
* GP8211S: 1-channel 15-bit VDAC with 0-5/10 V output range
* GP8403: 2-channel 12-bit VDAC with 0-5/10 V output range
* GP8413: 2-channel 15-bit VDAC with 0-5/10 V output range
* GP8302: 1-channel 12-bit IDAC with 0-25 mA output range

Although Guestgood is Chinese and doesn't offer English datasheets, small and well-documented development boards are produced by [DFRobot](https://www.dfrobot.com/), which are distributed by several known distributors such as Mouser and Digikey.

The 0-10V VDACs had my particular interest, because several home automation appliances can be controlled using 0-10V signals. These single-chip solutions are very ideal for this purpose.

### Testing procedure
A test application is provided in `tests/drivers/gp8xxx`. 

SAUL integration can be tested with the regular testing apps. Note that when using the SAUL shell, the numbers are parsed as Volts or Amperes. Unfortunately, the SAUL shell does not support fractional numbers, so testing the full range will be hard this way.

### Issues/PRs references
None

basilfx commented:
I took the liberty of rebasing this PR and adapt it to meet the conventions since last push.

Tested this once more using the SLSTK3401A with the GP8302 + GP8403 combination.

crasbe commented on drivers/gp8xxx/gp8xxx.c:
```diff
@@ -0,0 +1,198 @@
+#define GP8XXX_ADDR     (dev->params.address)
+
+#if GP8XXX_HAS_VDAC
+#define GP8XXX_RANGE    (dev->params.range)
```
```suggestion
#  define GP8XXX_RANGE  (dev->params.range)
```

crasbe commented on drivers/include/gp8xxx.h:
```diff
@@ -0,0 +1,221 @@
+ *
+ * The Guestgood GP8xxx I2C DACs are a series of digital-to-analog voltage
+ * (VDAC) and current (IDAC) converters. The driver supports a range of
+ * actuators, each with different number of channels and output ranges.
```
Not sure "actuators" is the word you'd want to use here? It's not a motor or anything like that 🤔 

Probably rather "variants" or "converters".

crasbe commented on drivers/include/gp8xxx.h:
```diff
@@ -0,0 +1,221 @@
+ * * GP8302: 1-channel 12-bit IDAC with 0-25 mA output range
+ *
+ * Storing output settings is not supported by this driver, because the driver
+ * will always initialize to a known state.
```
It's not clear to me what this means.

crasbe commented on drivers/include/gp8xxx.h:
```diff
@@ -0,0 +1,221 @@
+ * @brief Compile time macros to enable/disable features
+ *
+ * VDAC or IDAC support is only enabled if a module is selected that supports
+ * it. A Minor code size increase is expected when both a VDAC and IDAC module
```
```suggestion
 * it. A minor code size increase is expected when both a VDAC and IDAC modules
```
```suggestion
 * it. A Minor code size increase is expected when both a VDAC and IDAC module
```

crasbe commented on drivers/include/gp8xxx.h:
```diff
@@ -0,0 +1,221 @@
+ */
+typedef struct {
+    i2c_t i2c_dev;                  /**< I2C bus the actuator is connected to */
+    uint8_t address;                /**< actuator address */
```
Same as above regarding "actuator".

crasbe commented on drivers/include/gp8xxx.h:
```diff
@@ -0,0 +1,221 @@
+ *
+ * @return                  GP8XXX_OK on success
+ * @return                  GP8XXX_ERR_RANGE if value is out of range
+ * @return                  GP8XXX_ERR_I2C on I2C error
```
Same.

crasbe commented on tests/drivers/gp8xxx/README.md:
```diff
@@ -0,0 +1,13 @@
+
+# Usage
+This test application will initialize all DACs and provide a shell for
+interacting with the DACs
```
```suggestion
interacting with the DACs.
```

basilfx commented on dist/tools/doccheck/exclude_simple:
```diff
@@ -8405,6 +8405,26 @@ warning: Member LSM6DSL_REG_Y_OFS_USR (macro definition) of file lsm6dsxx_intern
+warning: Member GP8512_INFO_RANGE (macro definition) of file gp8xxx_info.h is not documented.
+warning: Member GP8512_INFO_RESOLUTION (macro definition) of file gp8xxx_info.h is not documented.
+warning: Member GP8XXX_REG_CHANNEL (macro definition) of file gp8xxx_constants.h is not documented.
+warning: Member GP8XXX_REG_OUTPUT (macro definition) of file gp8xxx_constants.h is not documented.
```
Documented them, so this commit will be removed.

basilfx commented:
@crasbe I have addressed all comments. I commented on the ones still unresolved, to check if you agree.

Did another round of testing with the SLSTK3401a (as in the videos) and it still works.

crasbe commented:
Here you go:
```
BOARD_INSUFFICIENT_MEMORY := \
    atmega8 \
    nucleo-l011k4 \
    #
```

basilfx commented:
Thank you @crasbe for the review and help to get this merged!
